### PR TITLE
Further password reset flow enhancements

### PR DESCRIPTION
**Pull request by weeman1337**
Designs here https://www.figma.com/file/C8zAfownE0T5pvyaJuwQK0/%5BLog-in%5D-Forgot-password

image

image

image

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Further password reset flow enhancements ([\#9662](https://github.com/matrix-org/matrix-react-sdk/pull/9662)).

### CODE REVIEW

**Comment by weeman1337** [comment]
Changes viewed by Percy are either unrelated or 1px off

File: src/components/structures/auth/ForgotPassword.tsx [hunk 1]
```diff
@@ -83,6 +86,10 @@ interface State {
 }
 
 export default class ForgotPassword extends React.Component<Props, State> {
+    public static defaultProps: Partial<Props> = {
+        onLoginClick: () => {},
```
Comment (robintown) [inline src/components/structures/auth/ForgotPassword.tsx:90]: Why is `onLoginClick` not a required prop?

File: res/css/views/dialogs/_VerifyEMailDialog.pcss [hunk 2]
```diff
@@ -34,5 +34,14 @@ limitations under the License.
             color: $secondary-content;
             line-height: 20px;
         }
+
+        .mx_AuthBody_did-not-receive {
+            justify-content: center;
+            margin-bottom: 8px;
```
Comment (robintown) [inline res/css/views/dialogs/_VerifyEMailDialog.pcss:40]:
```suggestion
            margin-bottom: $spacing-8;
```

File: res/css/views/auth/_AuthBody.pcss [hunk 3]
```diff
@@ -137,15 +137,50 @@ limitations under the License.
...
+    .mx_Login_submit {
+        font-weight: $font-semi-bold;
+        margin: 0 0 $spacing-16;
+    }
+
+    .mx_AuthBody_text {
+        margin-bottom: $spacing-32;
+
+        p {
+            margin: 0 0 8px;
```
Comment (robintown) [inline res/css/views/auth/_AuthBody.pcss:164]:
```suggestion
            margin: 0 0 $spacing-8;
```

**Review by robintown — COMMENTED** [review]
Generally looks good, one question

File: src/components/structures/auth/ForgotPassword.tsx (hunk 1 above)
Comment (weeman1337) [reply src/components/structures/auth/ForgotPassword.tsx:90]: Don't know thb. But every place using it passes `onLoginClick` as a prop → Made it required

**Comment by weeman1337** [comment]
Argh - I was too quick and rebased to `develop` instead of merging it :/ Sorry

**Review by robintown — APPROVED** [review]

**Comment by weeman1337** [comment]
Percy looks good. 1px shifts probably by rendering :man_shrugging:  or unrelated

**Pull request merged**